src/dpi.rs

```diff
@@ -1,80 +1,83 @@
+//! enter that state, they will no longer be focused on your application.
 //!
-//! That's a description of what happens when the button is 100x100 *physical* pixels. Instead, let's try using 100x100
-//! *logical* pixels. To map logical pixels to physical pixels, we simply multiply by the DPI (dots per inch) factor.
-//! On a "typical" desktop display, the DPI factor will be 1.0, so 100x100 logical pixels equates to 100x100 physical
-//! pixels. However, a 1440p display may have a DPI factor of 1.25, so the button is rendered as 125x125 physical pixels.
-//! Ideally, the button now has approximately the same perceived size across varying displays.
+//! ## How should I handle it?
 //!
-//! Failure to account for the DPI factor can create a badly degraded user experience. Most notably, it can make users
-//! feel like they have bad eyesight, which will potentially cause them to think about growing elderly, resulting in
-//! them entering an existential panic. Once users enter that state, they will no longer be focused on your application.
+//! The solution to this problem is to account for the device's *scale factor*. The scale factor is
+//! the factor UI elements should be scaled by to be consistent with the rest of the user's system -
+//! for example, a button that's normally 50 pixels across would be 100 pixels across on a device
+//! with a scale factor of `2.0`, or 75 pixels across with a scale factor of `1.5`.
 //!
-//! There are two ways to get the DPI factor:
-//! - You can track the [`HiDpiFactorChanged`](../enum.WindowEvent.html#variant.HiDpiFactorChanged) event of your
-//!   windows. This event is sent any time the DPI factor changes, either because the window moved to another monitor,
-//!   or because the user changed the configuration of their screen.
-//! - You can also retrieve the DPI factor of a monitor by calling
-//!   [`MonitorHandle::hidpi_factor`](../monitor/struct.MonitorHandle.html#method.hidpi_factor), or the
-//!   current DPI factor applied to a window by calling
-//!   [`Window::hidpi_factor`](../window/struct.Window.html#method.hidpi_factor), which is roughly equivalent
-//!   to `window.current_monitor().hidpi_factor()`.
+//! The scale factor correlates with, but no direct relationship to, the screen's actual DPI (dots
+//! per inch). Operating systems used to define the scale factor in terms of the screen's
+//! approximate DPI (at the time, 72 pixels per inch), but [Microsoft decided to report that the DPI
+//! was roughly 1/3 bigger than the screen's actual DPI (so, 96 pixels per inch) in order to make
+//! text more legible][microsoft_dpi]. As a result, the exact DPI as defined by the OS doesn't carry
+//! a whole lot of weight when designing cross-platform UIs. However, if you're in a situation where
+//! having an exact DPI value matters it should be safe to use `96 * scale_factor` on desktops and
+//! the web and `160 * scale_factor` on mobile platforms.
```

Comment:
This is a bit misleading as it ignores the fact that scale factor should account for the device's expected viewing distance as well, completely independent of the actual DPI.

Reply:
Could you elaborate on how this is misleading? This is mainly intended as a quick rule of thumb if you absolutely need some DPI value to plug into an equation or API that asks for it - not as a hard rule.

Reply:
I don't think it's appropriate to recommend a value that is expressly not DPI (e.g. wildly varying for the same panel depending on type of installation) be thought of as "kinda-sorta DPI", let alone used with APIs that require DPI. If the user wants to invent a plausible DPI based on scaling factor that's their prerogative, but it's not a generally reasonable thing to do with a scale factor.

Reply:
These aren't just values pulled out of thin air - the `96` is based on the DPI semantics for Windows, X11, and CSS, while `160` is the the documented base DPI factor for Android and (based on measurements) approximately the DPI factor for iOS. I'd be fine with making it more clear that these values aren't hard rules, but they'll be helpful numbers for most applications and as such I believe it's absolutely appropriate to include them.

Reply:
This information would not be helpful for any GUI application I've written or made meaningful contributions to. Android runs on both TVs and watches, whose DPI factors are going to be both wildly different *and* wildly inappropriate as the basis for user interface scaling. This paragraph encourages people to continue to think about user interface scaling in terms of DPI, which is wrong.

Reply:
I'm not arguing that the DPI should be used as the basis for user interface scaling. My intended usecase for that section is mainly for managing converting font sizes to pixel sizes , since font sizes are generally specified in points, rather than some DPI-agnostic unit. I'd assume that that information wouldn't have been useful to you since you'd likely have been using UI toolkits that automatically managed those conversions for you. Personally speaking, I can point to several specific times where I've been writing font handling code and would've liked to have a clearly documented DPI value so I could be certain that I was doing the conversions correctly.

These are the values that have been officially documented by OS authors and standard writers. This documentation isn't inventing anything new - it's just reporting the values that GUI toolkit designers have used as the bases for font scaling.

Reply:
Since winit has no obvious UI kit, we should be focused on docs that help folks build some good UI kit, and then consumers of that kit won't even read your docs since they'll read the kit's docs.

Reply:
Alright, I've done some more reading into HTML/CSS best practices, and I've updated the documentation to reflect that. Thank you for bringing this up.